src/commonMain/kotlin/fr/acinq/eclair/channel/Helpers.kt

```diff
@@ -363,6 +363,46 @@ object Helpers {
 
         fun isValidFinalScriptPubkey(scriptPubKey: ByteVector): Boolean = isValidFinalScriptPubkey(scriptPubKey.toByteArray())
 
+        fun btcAddressFromScriptPubKey(scriptPubKey: ByteVector, chainHash: ByteVector32): String? {
+            return runTrying {
+                val script = Script.parse(scriptPubKey)
+                when {
+                    Script.isPay2pkh(script) -> {
+                        // OP_DUP OP_HASH160 OP_PUSHDATA(20) OP_EQUALVERIFY OP_CHECKSIG
+                        val opPushData = script[2] as OP_PUSHDATA
+                        val prefix = when (chainHash) {
+                            Block.LivenetGenesisBlock.hash -> Base58.Prefix.PubkeyAddress
+                            Block.TestnetGenesisBlock.hash -> Base58.Prefix.PubkeyAddressTestnet
+                            else -> Base58.Prefix.PubkeyAddressSegnet
```

Comment:
`segnet` is not used anymore, regtest addresses also use the  `PubkeyAddressTestnet` prefix, and we should explicitly match against `Block.RegtestGenesisBlock`

Reply:
Done in cdeb6db